File: 07-langs_particularities/2-java/src/main/java/tv/codely/contexts/rrss/post_likes/domain/PostLike.java
package tv.codely.contexts.rrss.post_likes.domain;

public record PostLike(String id, String postId, String likerUserId) {
    public static PostLike like(String id, String postId, String likerUserId) {
        return new PostLike(id, postId, likerUserId);
    }
}

File: 07-langs_particularities/3-scala/src/main/scala/codely/contexts/rrss/post_likes/domain/PostLike.scala
package codely.contexts.rrss.post_likes.domain

import codely.contexts.shared.domain.DomainEvent

case class PostLikeDomainEvent() extends DomainEvent

enum PostLikeError:
	case PostLikeIdTooShort(id: String) extends PostLikeError

object PostLike {
	def fromPrimitives(id: String): Either[PostLikeError, PostLike] = Right(PostLike(id))
}

case class PostLike(id: String) {
	def pullDomainEvents(): Seq[PostLikeDomainEvent] = Seq()
}
